liquibase-standard/src/main/java/liquibase/changelog/ChangeSet.java

```diff
@@ -731,20 +737,22 @@ public ExecType execute(DatabaseChangeLog databaseChangeLog, ChangeExecListener
 
                 log.fine("Reading ChangeSet: " + this);
                 for (Change change : getChanges()) {
-                    if (listener != null) {
-                        listener.willRun(change, this, changeLog, database);
-                    }
-                    if (change.generateStatementsVolatile(database)) {
-                        executor.comment("WARNING The following SQL may change each run and therefore is possibly incorrect and/or invalid:");
-                    }
+                    if ((!(change instanceof DbmsTargetedChange)) || DatabaseList.definitionMatches(((DbmsTargetedChange) change).getDbms(), database, true)) {
```

Comment:
Moved this check back here (before 4.22.0 it was here and not at DbmsChangeSetFilter)